arbor/network.cpp

```diff
@@ -0,0 +1,1551 @@
+#include <arbor/common_types.hpp>
+#include <arbor/network.hpp>
+
+#include <Random123/threefry.h>
+#include <Random123/boxmuller.hpp>
```

Comment:
The algorithm should not change, since this would impact reproducibility. And beyond types, I don't see much in common with other usage of random123 elsewhere in the code to add a layer in between. Did you have something in mind for this layer that goes beyond types?

Reply:
Our use of R123 has grown and I have an open issue to use it in `schedule`, too. (#2243)  
Thus, I'll use that opportunity to consolidate and make some thin abstractions.